### PR TITLE
feat(components): add public re export pattern

### PR DESCRIPTION
**Pull request by aklkv**
### :pushpin: Summary

* add public-facing import pattern for importing components/helpers/modifiers

### :link: External links

Jira ticket: [HDS-XXX](https://hashicorp.atlassian.net/browse/HDS-XXX)
Figma file: [if it applies]

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/src/components.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import HdsAppFrame from './components/hds/app-frame/index.ts';
+import HdsCardContainer from './components/hds/card/container.ts';
+import HdsDisclosurePrimitive from './components/hds/disclosure-primitive/index.ts';
+import HdsDismissButton from './components/hds/dismiss-button/index.ts';
+import HdsIconTile from './components/hds/icon-tile/index.ts';
+import HdsInteractive from './components/hds/interactive/index.ts';
+import HdsLinkInline from './components/hds/link/inline.ts';
+import HdsLinkStandalone from './components/hds/link/standalone.ts';
+import HdsReveal from './components/hds/reveal/index.ts';
+import HdsRevealToggleButton from './components/hds/reveal/toggle/button.ts';
```
Comment (alex-ju) [inline packages/components/src/components.ts:18]: I believe we can discard `HdsRevealToggleButton` as it's a subcomponent of `HdsReveal`

File: packages/components/src/components.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  HdsAppFrame,
+  HdsCardContainer,
+  HdsDisclosurePrimitive,
+  HdsDismissButton,
+  HdsIconTile,
+  HdsInteractive,
+  HdsLinkInline,
+  HdsLinkStandalone,
+  HdsReveal,
+  HdsRevealToggleButton,
```
Comment (alex-ju) [inline packages/components/src/components.ts:44]: Same here, `HdsRevealToggleButton` can be discarded

**Comment by alex-ju** [comment]
I rebased this PR to add a few entries for the components merged in the meantime, double-checked to make sure we have all main components added to the list, added a changelog entry, and updated the TypeScript migration docs to cover this change.

@didoo when time allows, would you be able to have a look?

**Review by didoo — APPROVED** [review]
👍

**Pull request merged**